src/components/pages/donation_form/DonationSummary.vue

```diff
@@ -1,60 +1,63 @@
-import PaymentSummaryPrivate from '@src/components/pages/donation_form/PaymentSummaryContent/PaymentSummaryPrivate.vue';
-import { Country } from '@src/view_models/Country';
-import { Salutation } from '@src/view_models/Salutation';
 import { useI18n } from 'vue-i18n';
 import { computed } from 'vue';
+import DonorSummary from '@src/components/pages/donation_form/DonorSummary.vue';
+import type { Address } from '@src/view_models/Address';
+import type { Country } from '@src/view_models/Country';
+import type { Salutation } from '@src/view_models/Salutation';
 
 interface Props {
-	address: Record<string, string>;
-	addressType: String;
+	address: Address;
 	payment: { interval: any; amount: number; paymentType: any };
+	bankData: { iban: string; bic: string; bankName: string };
 	countries: Array<Country>;
-	languageItem: String;
 	salutations: Array<Salutation>;
 }
 
 const { t, n } = useI18n();
 const props = defineProps<Props>();
 
-const addressTypeComponents = [
-	{ key: addressTypeName( AddressTypeModel.ANON ), component: PaymentSummaryAnonymous },
-	{ key: addressTypeName( AddressTypeModel.COMPANY ), component: PaymentSummaryCompany },
-	{ key: addressTypeName( AddressTypeModel.COMPANY_WITH_CONTACT ), component: PaymentSummaryCompanyWithContact },
-	{ key: addressTypeName( AddressTypeModel.EMAIL ), component: PaymentSummaryEmail },
-	{ key: addressTypeName( AddressTypeModel.PERSON ), component: PaymentSummaryPrivate },
-	{ key: addressTypeName( AddressTypeModel.UNSET ), component: PaymentSummaryAnonymous },
-];
-
-// TODO: Extract this into a composable
-const addressTypeComponent = computed( () => addressTypeComponents.find( x => x.key === props.addressType ).component );
-const paymentType = computed( () => t( props.payment.paymentType ) );
-const interval = computed( () => t( 'donation_form_payment_interval_' + props.payment.interval ) );
 const formattedAmount = computed( () => n( props.payment.amount, { key: 'currency', currencyDisplay: 'name' } ) );
-const country = computed( () => {
-	const countryObject = props.countries.find( c => ( c.countryCode === props.address.countryCode || c.countryCode === props.address.country ) );
-	return countryObject ? countryObject.countryFullName : '';
-} );
-const salutation = computed( () => props.address.salutation ? props.salutations.find( s => s.value === props.address.salutation )?.display : '' );
 
+const interval = computed( () => props.payment.interval ? t( 'donation_form_payment_interval_' + props.payment.interval ) : '' );
+
+const paymentType = computed( () => props.payment.paymentType ? t( props.payment.paymentType ) : '' );
 </script>
+
+<style lang="scss">
+@use '@src/scss/settings/units';
+@use 'sass:map';
+
+.switcher {
+	display: flex;
+	flex-wrap: wrap;
+	gap: map.get( units.$spacing, 'small' );
+}
+
+.switcher > * {
+	flex-grow: 1;
+	flex-basis: calc((40rem - 100%) * 999);
```

Comment:
You need to fiddle with the `40rem` value here so the children will be displayed in a row when there's space. `26rem` seems like a good value.